src/bun.js/node/net/BlockList.zig

```diff
@@ -0,0 +1,249 @@
+                    const set_adr = std.bit_set.IntegerBitSet(32){ .mask = @byteSwap(@bitReverse(a_r)) };
+                    const intersection = set_net.xorWith(set_adr);
+                    const t = @ctz(intersection.mask);
+                    const h = t >= s.prefix;
+                    if (h) return .jsBoolean(true);
+                };
+                if (address.sin.family == std.posix.AF.INET6 and s.network.sin.family == std.posix.AF.INET6) {
+                    const set_net = std.bit_set.IntegerBitSet(128){ .mask = @byteSwap(@bitReverse(@as(u128, @bitCast(s.network.sin6.addr)))) };
+                    const set_adr = std.bit_set.IntegerBitSet(128){ .mask = @byteSwap(@bitReverse(@as(u128, @bitCast(address.sin6.addr)))) };
+                    const intersection = set_net.xorWith(set_adr);
+                    const t = @ctz(intersection.mask);
+                    const h = t >= s.prefix;
+                    if (h) return .jsBoolean(true);
+                }
+            },
+        }
+    }
+    return .jsBoolean(false);
+}
+
+pub fn rules(this: *@This(), globalThis: *JSC.JSGlobalObject) JSC.JSValue {
+    this.mutex.lock();
+    defer this.mutex.unlock();
+    var list = std.ArrayList(JSC.JSValue).initCapacity(bun.default_allocator, this.da_rules.items.len) catch bun.outOfMemory();
+    defer list.deinit();
+    for (this.da_rules.items) |rule| {
+        switch (rule) {
+            .addr => |a| {
+                var buf: [SocketAddress.inet.INET6_ADDRSTRLEN]u8 = @splat(0);
+                list.appendAssumeCapacity(bun.String.createFormatForJS(globalThis, "Address: {s} {s}", .{ a.family().upper(), a.fmt(&buf) }));
+            },
+            .range => |r| {
+                var buf_s: [SocketAddress.inet.INET6_ADDRSTRLEN]u8 = @splat(0);
+                var buf_e: [SocketAddress.inet.INET6_ADDRSTRLEN]u8 = @splat(0);
+                list.appendAssumeCapacity(bun.String.createFormatForJS(globalThis, "Range: {s} {s}-{s}", .{ r.start.family().upper(), r.start.fmt(&buf_s), r.end.fmt(&buf_e) }));
+            },
+            .subnet => |s| {
+                var buf: [SocketAddress.inet.INET6_ADDRSTRLEN]u8 = @splat(0);
+                list.appendAssumeCapacity(bun.String.createFormatForJS(globalThis, "Subnet: {s} {s}/{d}", .{ s.network.family().upper(), s.network.fmt(&buf), s.prefix }));
+            },
+        }
+    }
+    return JSC.JSArray.create(globalThis, list.items);
+}
+
+pub fn onStructuredCloneSerialize(this: *@This(), globalThis: *JSC.JSGlobalObject, ctx: *anyopaque, writeBytes: *const fn (*anyopaque, ptr: [*]const u8, len: u32) callconv(JSC.conv) void) void {
+    _ = globalThis;
+    this.mutex.lock();
+    defer this.mutex.unlock();
+    this.ref();
+    const writer = StructuredCloneWriter.Writer{ .context = .{ .ctx = ctx, .impl = writeBytes } };
+    try writer.writeInt(usize, @intFromPtr(this), .little);
```

Comment:
We're serializing the pointer address? What happens when the process restarts?

Reply:
for this script bun and node are identical 

```ts
const assert = require("assert");
const net = require("net");
const cluster = require("cluster");

if (cluster.isPrimary) {
  const worker = cluster.fork({});

  worker.on("message", function (data) {
    const blocklist = data.blocklist;
    console.log("primary", blocklist); // primary {}
    // `blocklist` is empty object so cannot call .addAddress() again
    process.exit(0);
  });
} else {
  const blocklist = new net.BlockList();
  blocklist.addAddress("123.123.123.123");

  console.log("worker", blocklist); // worker BlockList { rules: [ 'Address: IPv4 123.123.123.123' ] }
  const result = cluster.worker.send({ blocklist });
  assert.strictEqual(result, true);
}
```

Reply:
this was not using advanced serialization, it does crash